File: src/java/_2682_Find_the_Losers_of_the_Circular_Game.java
class Solution {
    public int[] circularGameLosers(int n, int k) {
        boolean seen[] = new boolean[n];

        int i = 1, num = 0;

        while (!seen[num]) {
            seen[num] = true;
            num += i * k;
            num %= n;
            i++;
        }

        int[] ans = new int[n - i + 1];
        i = 0;

        for (int j = 0; j < n; j++)
            if (!seen[j])
                ans[i++] = j + 1;

        return ans;
    }
}

File: src/python/_2682_Find_the_Losers_of_the_Circular_Game.py
class Solution:
    def circularGameLosers(self, n: int, k: int) -> List[int]:
        seen = set()
        i = 1
        num = 0

        while num not in seen:
            seen.add(num)
            num += i * k
            num %= n
            i += 1

        ans = []

        for num in range(n):
            if num not in seen:
                ans.append(num + 1)

        return ans
